dotCMS/src/integration-test/java/com/dotmarketing/quartz/DotStatefulJobTest.java

```diff
@@ -0,0 +1,188 @@
+package com.dotmarketing.quartz;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.quartz.DotStatefulJob.TriggerBuilder;
+import com.dotmarketing.quartz.job.TestJobExecutor;
+import java.util.UUID;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.CyclicBarrier;
+import java.util.concurrent.TimeUnit;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.quartz.JobDataMap;
+import org.quartz.JobDetail;
+import org.quartz.Scheduler;
+import org.quartz.SchedulerException;
+import org.quartz.Trigger;
+import org.quartz.core.SchedulingContext;
+
+/**
+ * Tests used to verify that {@link DotStatefulJob} instances only create one trigger for multiple
+ * jobs
+ *
+ * @author nollymar
+ */
+public class DotStatefulJobTest {
+
+    private final static int THREAD_AMOUNT = 7;
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```

Comment:
Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5971105)